src/components/CreateBoard/CreateBoard.jsx

```diff
@@ -0,0 +1,272 @@
+import React from "react";
+import css from './CreateBoard.module.css';
+import {customHistory} from "../../App";
+import InputErrorValidation from "../../images/InputErrorValidationImage";
+import {ErrorMessage} from "../Errors/ErrorMessage/ErrorMessage";
+import {Preloader} from "../Preloader/Preloader";
+
+export class CreateBoard extends React.Component {
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            formGet: {
+                category: [],
+                icon: []
+            },
+            formPost: {
+                title: '',
+                key: '',
+                category: {
+                    key: '',
+                    value: ''
+                },
+                icon: {
+                    key: '',
+                    value: ''
+                }
+            },
+            errorValidation: {
+                title: '',
+                key: '',
+            },
+            errorMessage: '',
+            responseStatus: ''
+        }
+    }
+
+
+    componentDidMount() {
+        fetch('/dictionaries/categories')
+            .then(response => response.json())
+            .then(response => {
+                this.setState({
+                    formGet: {
+                        ...this.state.formGet,
+                        category: response
+                    }
+                });
+            })
+            .catch(error => {
+                this.setState({errorMessage: error.message});
+            });
+
+        fetch('/dictionaries/board-icons')
+            .then(response => response.json())
+            .then(response => {
+                this.setState({
+                    formGet: {
+                        ...this.state.formGet,
+                        icon: response
+                    }
+                });
+                this.props.isFetching(false);
+            })
+            .catch(error => {
+                this.setState({errorMessage: error.message});
+                this.props.isFetching(false);
+            });
+    }
+
+    onChange = (event) => {
+        if (event) {
+            event.preventDefault();
+        }
+
+        this.setState({
+            formPost: {
+                ...this.state.formPost,
+                [event.target.name]: event.target.value
+            }
+        });
+    };
+
+    onChangeCategoryAndIcon = (event) => {
+        if (event) {
+            event.preventDefault();
+        }
+
+        if (event.target.name === "category") {
+            this.setState({
```

Comment:
I think it might replaced with:
```
this.setState({
                formPost: {
                    ...this.state.formPost,
                    category: this.state.formGet.category.find(
                        category => category.value === event.target.value
                    )
                }
            })
```
But better to check

Reply:
done